lib/tasks/puppet.rake

```diff
@@ -57,4 +57,36 @@ namespace :puppet do
       end
     end
   end
+
+  namespace :migrate do
+    desc "
+    Migrate Puppet provisioning parameters from ENC to Host parameters.
+    This task is relevant if you want to continue using Puppet provisioning,
+    but do not want to install the Puppet plugin and get the full ENC funcionality."
+
+    task :to_parameters => :environment do
+      class FakeEnv < ::ApplicationRecord
+        self.table_name = 'environments'
+      end
+
+      User.current = User.anonymous_console_admin
+      envs = Hash[FakeEnv.all.pluck(:id, :name)]
+
+      hg_at = Hostgroup.arel_table
+      hg_arel = hg_at[:environment_id].not_eq(nil)
+      hg_arel = hg_arel.or(hg_at[:puppet_proxy_id].not_eq(nil))
+      hg_arel = hg_arel.or(hg_at[:puppet_ca_proxy_id].not_eq(nil))
+      Hostgroup.where(hg_arel).find_each(batch_size: 100) do |hg|
+        GroupParameter.create(hostgroup: hg, name: 'puppet_environment', value: envs[hg.environment_id])
+        GroupParameter.create(hostgroup: hg, name: 'puppet_server', value: hg.puppet_server)
+        GroupParameter.create(hostgroup: hg, name: 'puppet_ca_server', value: hg.puppet_ca_server)
+      end
+
+      Host.where.not(environment_id: nil).find_each(batch_size: 100) do |host|
+        HostParameter.create(host: host, name: 'puppet_environment', value: envs[host.environment_id])
+        HostParameter.create(host: host, name: 'puppet_server', value: host.puppet_server)
+        HostParameter.create(host: host, name: 'puppet_ca_server', value: host.puppet_ca_server)
```

Comment:
I don't remember if it's possible to only set some of these three, but if it is than the comment above also applies here.